src/lightning/pytorch/core/datamodule.py

```diff
@@ -157,7 +158,7 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         """
         pass
 
-    @classmethod
+    @_restricted_classmethod
```

Comment:
Is the Self return annotation still correct now that we've changed the decorator?
```
src/lightning/pytorch/core/module.py:1550: error: Argument 1 to "_load_from_checkpoint" has incompatible type "Self"; expected "type[LightningModule] | type[LightningDataModule]"  [arg-type]
src/lightning/pytorch/core/datamodule.py:228: error: Argument 1 to "_load_from_checkpoint" has incompatible type "Self"; expected "type[LightningModule] | type[LightningDataModule]"  [arg-type]

```

Should we ignore the typing error or change the type?

Reply:
Ignore it IMO